rerun_py/tests/e2e_redap_tests/conftest.py

```diff
@@ -11,41 +11,67 @@
 import pathlib
 import platform
 import re
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
 import pyarrow as pa
 import pytest
-from rerun.catalog import CatalogClient
+from rerun.catalog import CatalogClient, DatasetEntry, TableEntry
 from rerun.server import Server
 from syrupy.extensions.amber import AmberSnapshotExtension
 
 if TYPE_CHECKING:
     from collections.abc import Generator
 
-    from rerun.catalog import DatasetEntry, TableEntry
+    from pytest_benchmark.fixture import BenchmarkFixture
     from syrupy import SnapshotAssertion
 
 
 def pytest_addoption(parser: pytest.Parser) -> None:
     """Add custom command-line options for configuring the test server."""
+
     parser.addoption(
+        "--resource-prefix",
+        action="store",
+        default=None,
+        help="URI prefix for test resources (e.g., 's3://bucket/path/' for remote resources). "
+        "If not provided, local file:// URIs to the resources directory will be used.",
+    )
+    parser.addoption(
+        "--region",
+        action="store",
+        default="None",
+        help="region to use when using region-specific resources (e.g., s3 buckets).",
+    )
+
+    serveropts_group = parser.getgroup("redap server options")
+    serveropts_group.addoption(
         "--redap-url",
         action="store",
         default=None,
         help="URL of an external redap server to connect to. If not provided, a local OSS server will be started.",
     )
-    parser.addoption(
+    serveropts_group.addoption(
         "--redap-token",
         action="store",
         default=None,
         help="Authentication token for the redap server (optional).",
     )
-    parser.addoption(
-        "--resource-prefix",
+
+    # Create a specific group in the help menu so they aren't mixed with generic flags
+    droid_group = parser.getgroup("droid dataset options")
+
+    droid_group.addoption(
+        "--droid-dataset",
         action="store",
-        default=None,
-        help="URI prefix for test resources (e.g., 's3://bucket/path/' for remote resources). "
-        "If not provided, local file:// URIs to the resources directory will be used.",
+        default="droid:sample50",
+        help="Dataset for Droid benchmarks (options: droid:raw, droid:sample50, etc.)",
+    )
+
+    droid_group.addoption(
+        "--droid-preregister-dataset",
+        action="store_true",
+        default=False,
+        help="should the test runner register the droid dataset before running tests",
     )
```

Comment:
It would be slightly more readable if the `adoption` code was colocated with the `getoption` fixtures. Would it make sense to move that to the other conftest.py?